util/Pos/Util/Verification.hs

```diff
@@ -0,0 +1,101 @@
+{- |
+
+Machinery to support pure datatype verification.
```

Comment:
IIRC you were going to enumerate advantages of this machinery (as I requested), but I still don't see them.

Reply:
Added.